resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/dtree/JPath.java

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Joerg Kubitz and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Joerg Kubitz - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.core.internal.dtree;
+
+import java.util.Arrays;
+import java.util.Objects;
+import org.eclipse.core.runtime.IPath;
+
+public record JPath(String[] segments) {
```

Comment:
Can you please add javadoc that explains why/what this record is used for.

Reply:
done